src/library_syscall.js

```diff
@@ -1046,13 +1046,18 @@ var SyscallsLibrary = {
     assert(flags === 0);
 #endif
     path = SYSCALLS.calculateAt(dirfd, path, true);
-    var seconds = {{{ makeGetValue('times', C_STRUCTS.timespec.tv_sec, 'i32') }}};
-    var nanoseconds = {{{ makeGetValue('times', C_STRUCTS.timespec.tv_nsec, 'i32') }}};
-    var atime = (seconds*1000) + (nanoseconds/(1000*1000));
-    times += {{{ C_STRUCTS.timespec.__size__ }}};
-    seconds = {{{ makeGetValue('times', C_STRUCTS.timespec.tv_sec, 'i32') }}};
-    nanoseconds = {{{ makeGetValue('times', C_STRUCTS.timespec.tv_nsec, 'i32') }}};
-    var mtime = (seconds*1000) + (nanoseconds/(1000*1000));
+    if (!times) {
+      var atime = Date.now()/1000;
+      var mtime = atime;
```

Comment:
Seems a little odd to declare `var mtime` and `var atime` in both branches of the if/else.   Maybe just declare them in one?  (I'm not a JS expert but IIRC you just need one `var` decl for a give variable it it will be hoisted to function level).

Reply:
I don't know any JS at all. :)

Reply:
`mtime` and `atime` should be declared on level up (`var mtime, atime;`).
Otherwise the `FS.utime(path, atime, mtime)` will not be able to access the variables on the scope.

Reply:
Is that true?  I thought `var` declaration were hoisted to the function level in JS.  This seems to work just fine:

```
function foo() {                                                                 
  if (0) {                                                                          
    var bar = 10;                                                                   
  } else {                                                                       
    bar = 7;                                                                     
  }                                                                                                
  console.error(bar);                                                            
} 
```

It looks odd but its saves a few bytes in code size.

Reply:
You are completely right! I got a bit confused with var and let :P